specs/Standards/MultiToken/Events.md

```diff
@@ -0,0 +1,271 @@
+Events are log entries that start with `EVENT_JSON:` prefix followed by a single valid JSON document of the following interface:
+
+```ts
+// Interface to capture data
+// about an event
+// Arguments
+// * `standard`: name of standard e.g. nep171
+// * `version`: e.g. 1.0.0
+// * `event`: string
+// * `data`: associate event data
+interface EventLogData {
+    standard: string,
+    version: string,
+    event: string,
+    data?: unknown,
+}
+```
+
+#### Valid event logs:
+
+```js
+EVENT_JSON:{"standard": "nepXXX", "version": "1.0.0", "event": "xyz_is_triggered"}
+```
+
+```js
+EVENT_JSON:{
+  "standard": "nepXXX",
+  "version": "1.0.0",
+  "event": "xyz_is_triggered"
+}
+```
+
+```js
+EVENT_JSON:{"standard": "nepXXX", "version": "1.0.0", "event": "xyz_is_triggered", "data": {"triggered_by": "foundation.near"}}
+```
+
+#### Invalid event logs:
+
+* Two events in a single log entry (instead, call `log` for each individual event)
+```
+EVENT_JSON:{"standard": "nepXXX", "version": "1.0.0", "event": "xyz_is_triggered"}
+EVENT_JSON:{"standard": "nepXXX", "version": "1.0.0", "event": "xyz_is_triggered"}
+```
+* Invalid JSON data
+```
+EVENT_JSON:invalid json
+```
+
+
+## Interface
+Multi Token Events MUST have `standard` set to `"nep246"`, standard version set to `"1.0.0"`, `event` value is one of `mt_mint`, `mt_burn`, `mt_transfer`, `mt_approval` and `data` must be of one of the following relavant types: `MtMintLog[] | MtBurnLog[] | MtTransferLog[] | MtApprovalLog[]`:
+
+
+
+```ts
+// Interface to capture an event. The indexer looks for events prefixed 
+// with EVENT_JSON. The rest follows the EventLog 
+type MtEvent = "mt_mint" | "mt_burn" | "mt_transfer" | "mt_approval"
+
+// Interface for MT contract event data. It is used to emit event data with
+// the near standard logging.    
+// * `EVENT_JSON`: The standard event prefix required to signal consumers about
+// the type of log data being emitted.
+// * `standard`: name of standard e.g. nep-246 
+// * `version`: e.g. "1.0.0"
+// * `event`: `mt_mint` | `mt_burn` | `mt_transfer` | `mt_approval`
+// * `data`: associate event data
+interface MtEventLogData {
+  EVENT_JSON: {
+    standard: "nep246",
+    version: "1.0.0",
+    event: MtEvent,
+    data: MtMintLog[] | MtBurnLog[] | MtTransferLog[] | MtApprovalLog[]
+  }
+}
+
+// Minting event log. Emitted when a token is minted/created. 
+// Requirements
+// * Contract MUST emit event when minting a token
+// Fields 
+// * Contract token_ids and amounts MUST be the same length 
+// * `owner_id`: the account receiving the minted token
+// * `token_ids`: the tokens minted
+// * `amounts`: the number of tokens minted, wrapped in quotes and treated
+//    like a string, although the numbers will be stored as an unsigned integer
+//    array with 128 bits.
+// * `memo`: optional message
+interface MtMintLog {
+    owner_id: string,
+    token_ids: string[],
+    amounts: string[],
```

Comment:
Here and in all other logs:

These arrays should be the same length. Users could mess that up.

I'd prefer to have 1 array of jsons, something like
```
tokens: [
    {token_id: "1", amount: "5"},
    {token_id: "awesome", amount: "10"},
]
```
But I guess it's too wordy.
Maybe we can create a third idea of how to work with that in a more elegant way?

Reply:
An alternative could be a couple of the data. I think that's maybe harder to communicate though.
MtMintLog {
owner_id: string
tokens: [ ["1","5"], ["awesome","10"]]
}

I think in terms of user familiarity with the restrictions, there's already a precedent set from the core implementations, They all have the same restrictions. These restrictions exist in the mirror standard 1155 as well . So users shouldn't be blindsided by it, and in most situations I think they could just pass the data along. 🤔